pkg/scheduler/preemption/preemption.go

```diff
@@ -543,15 +539,9 @@ func cqIsBorrowing(cq *cache.ClusterQueueSnapshot, resPerFlv resourcesPerFlavor)
 	if cq.Cohort == nil {
 		return false
 	}
```

Comment:
can we get rid of this check? if so, can fold `queueUnderNominalInResourcesNeedingPreemption` into this function. unit tests still pass after its removal

Reply:
Yeah, I think the check can be removed, because this logic is only called for CQs that belong to the same cohort as the preempting one.

Reply:
Replaced usages of `queueUnderNominalInResourcesNeedingPreemption` with `!cqIsBorrowing`